### PR TITLE
ISSUE-27500: Fixed unit tests for module Magento_SampleData

### PR DESCRIPTION
**Pull request by andrewbess**
### Description (*)

This PR partially updates unit tests in SampleData module
Other tests of this module have been updated in https://github.com/magento/magento2/pull/27481

### Fixed Issues (if relevant)

1. magento/magento2#27500: Unit Tests incompatible with PHPUnit 8

### Manual testing scenarios (*)

1. Run tests in phpunit 6 (currently in 2.4-develop) using `vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist app/code/Magento/SampleData/`
2. Install phpunit 8, run tests using `vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist app/code/Magento/SampleData/`

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/SampleData/Test/Unit/Model/DependencyTest.php [hunk 1]
```diff
@@ -12,41 +14,50 @@
...
+use Magento\Framework\Filesystem\Directory\ReadInterface;
 use Magento\Framework\Phrase;
 use Magento\SampleData\Model\Dependency;
 use Magento\Framework\Filesystem\DriverPool;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
 
 /**
- * Class DependencyTest
+ * Tests for Magento\SampleData\Model\Dependency
```
Comment (lbajsarowicz) [inline app/code/Magento/SampleData/Test/Unit/Model/DependencyTest.php]: If you really want to provide the class - you can use `@covers` annotation

File: app/code/Magento/SampleData/Test/Unit/Model/DependencyTest.php [hunk 2]
```diff
@@ -197,11 +210,11 @@ public function stubComposerJsonReader(array $composerJsonContent)
     }
 
     /**
-     * @return \PHPUnit_Framework_MockObject_MockObject
+     * @return MockObject
```
Comment (lbajsarowicz) [inline app/code/Magento/SampleData/Test/Unit/Model/DependencyTest.php]:
```suggestion
     * @return ReadInterface|MockObject
```

File: app/code/Magento/SampleData/Test/Unit/Model/DependencyTest.php [hunk 3]
```diff
@@ -178,11 +190,12 @@ public static function dataPackagesFromComposerSuggest()
 
     /**
      * @param array $composerJsonContent
-     * @return \PHPUnit_Framework_MockObject_MockObject
+     * @return MockObject
```
Comment (lbajsarowicz) [inline app/code/Magento/SampleData/Test/Unit/Model/DependencyTest.php]:
```suggestion
     * @return ReadInterface|MockObject
```

File: app/code/Magento/SampleData/Test/Unit/Model/DependencyTest.php [hunk 4]
```diff
@@ -178,11 +190,12 @@ public static function dataPackagesFromComposerSuggest()
...
      */
-    public function stubComposerJsonReader(array $composerJsonContent)
-    {
-        $stub = $this->getMockBuilder(Filesystem\Directory\ReadInterface::class)
+    public function stubComposerJsonReader(
+        array $composerJsonContent
+    ): MockObject {
```
Comment (lbajsarowicz) [inline app/code/Magento/SampleData/Test/Unit/Model/DependencyTest.php]:
```suggestion
    public function stubComposerJsonReader(array $composerJsonContent): MockObject {
```

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Please review the comments. Only minor changes to be introduced.

**Review by lbajsarowicz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7263](https://jira.corp.magento.com/browse/ENGCOM-7263) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Review by lbajsarowicz — DISMISSED** [review]
:heavy_check_mark: Great job, thank you for your contribution.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7263](https://jira.corp.magento.com/browse/ENGCOM-7263) has been created to process this Pull Request

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @andrewbess 

During testing, we faced the issue

PHPUnit 8

**Run:** `vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist app/code/Magento/SampleData/`

**Actual Result:**
Screenshot from 2020-04-09 12-20-54

@andrewbess Could you take a look? 
Thanks!

**Comment by andrewbess** [comment]
> -c dev/tests/unit/phpunit.xml.dist

Hello @engcom-Alfa 
It has already done earlier in https://github.com/magento/magento2/pull/27481 but still doesn't merged.

**Comment by slavvka** [comment]
@andrewbess Please resolve the conflict

**Comment by andrewbess** [comment]
Hello @slavvka 
Done.
Please check.
Thank you in advance.

File: app/code/Magento/SampleData/Test/Unit/Model/DependencyTest.php [hunk 5]
```diff
@@ -14,30 +16,36 @@
...
 use Magento\Framework\Phrase;
 use Magento\SampleData\Model\Dependency;
+use Magento\Framework\Filesystem\DriverPool;
 use PHPUnit\Framework\MockObject\MockObject;
 use PHPUnit\Framework\TestCase;
 
 /**
+ * Provides tests for Dependency model of SampleData module
+ *
+ * @covers \Magento\SampleData\Model\Dependency
```
Comment (lbajsarowicz) [inline app/code/Magento/SampleData/Test/Unit/Model/DependencyTest.php:28]: It's not crucial to provide that, but it's OK if you put effort on adding it :-)

**Review by lbajsarowicz — APPROVED** [review]
Looks fine for me.

**Review by slavvka — APPROVED** [review]

**Pull request merged**